src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -607,40 +604,37 @@ private class StatusBarProgress : RestoreOperationProgressUI
                 JoinableTaskFactory jtf,
                 CancellationToken token)
             {
-                return await jtf.RunAsync(async () =>
-                {
-                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                await jtf.SwitchToMainThreadAsync();
 
-                    var statusBar = serviceProvider.GetService<SVsStatusbar, IVsStatusbar>();
+                var statusBar = serviceProvider.GetService<SVsStatusbar, IVsStatusbar>();
 
-                    // Make sure the status bar is not frozen
-                    int frozen;
-                    statusBar.IsFrozen(out frozen);
+                // Make sure the status bar is not frozen
+                int frozen;
+                statusBar.IsFrozen(out frozen);
 
-                    if (frozen != 0)
-                    {
-                        statusBar.FreezeOutput(0);
-                    }
+                if (frozen != 0)
+                {
+                    statusBar.FreezeOutput(0);
+                }
 
-                    statusBar.Animation(1, ref icon);
+                statusBar.Animation(1, ref Icon);
```

Comment:
Synchronously blocking scenarios would presumably just use TWD rather than status bar, I think.
Async scenarios have used status bar in the past, but we're encouraging folks to switch to the Task Status Center more recently. TSC will actually drive the status bar text and progress bar in the future. Just something to keep in mind for the future.